jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -205,7 +222,12 @@ def mos_spec1d_parser(app, data_obj, data_labels=None,
     if not isinstance(data_obj, (list, tuple, SpectrumCollection)):
         data_obj = [data_obj]
 
-    data_obj = [Spectrum1D.read(x) if _check_is_file(x) else x for x in data_obj]
+    try:
```

Comment:
Please comment in the code of the *intent* here. It is not clear to me why this try-except is necessary.